registry/remote/auth/scope.go

```diff
@@ -93,7 +119,62 @@ func AppendScopes(ctx context.Context, scopes ...string) context.Context {
 // GetScopes returns the scopes in the context.
 func GetScopes(ctx context.Context) []string {
 	if scopes, ok := ctx.Value(scopesContextKey{}).([]string); ok {
-		return append([]string(nil), scopes...)
+		return slices.Clone(scopes)
+	}
+	return nil
+}
+
+// scopesPerHostContextKey is the context key for per-host scopes.
+type scopesPerHostContextKey struct{}
```

Comment:
Why not make
```suggestion
type scopesPerHostContextKey string
```
where the `string` stores `host`?

Reply:
so that we don't require a map

Reply:
> Why not make
> 
> where the `string` stores `host`?

According to the [doc](https://pkg.go.dev/context#WithValue:~:text=The%20provided%20key%20must%20be%20comparable%20and%20should%20not%20be%20of%20type%20string%20or%20any%20other%20built%2Din%20type%20to%20avoid%20collisions%20between%20packages%20using%20context.), we should not use a string as context key.

Reply:
Just realized that context key can actually be a `string` as long as it's a custom type so that there won't be collisions between packages, as shown in the [example](https://pkg.go.dev/context#example-WithValue).

Reply:
Updated.